src/main/java/hudson/plugins/git/util/GitUtils.java

```diff
@@ -190,7 +219,11 @@ public Revision sortBranchesForRevision(Revision revision, List<BranchSpec> bran
                         }
 
                     } finally {
-                        walk.release();
+                        try {
+                            walk.release();
```

Comment:
In other situations I have had a utility class with a lambda field initialized lazily. The lazy init determines which impl to use so that the JVM can inline all the way (the JVM will optimise reflection to direct method call if you give it a definitive constant case) but that is normally only needed for hot codepaths. 

Given this code path seems non heavy, I'm unsure if it is necessary to go the 

```
public class WalkerUtils {
  private static Lambda closer = create();

  public void close(walker) {
    closer.close(walker);
  }

  public Lambda create() {
    try {
      Walker.class.getMethod("close");
      return new ReflectionLambda();
    } catch (NoSuchMethod e) {
      return new JGit3Lambda();
    }
  }

  ...

```